packages/Chem/src/analysis/molecular-matched-pairs/mmp-pairs-transforms.ts

```diff
@@ -0,0 +1,236 @@
+import * as DG from 'datagrok-api/dg';
+import {ILineSeries} from '@datagrok-libraries/utils/src/render-lines-on-sp';
+import BitArray from '@datagrok-libraries/utils/src/bit-array';
+import {MmpRules} from './mmp-fragments';
+import {SUBSTRUCT_COL} from '../../constants';
+import {MMP_COLNAME_FROM, MMP_COLNAME_TO, MMP_COLNAME_PAIRS, MMP_COL_PAIRNUM, MMP_COL_PAIRNUM_FROM,
+  MMP_COL_PAIRNUM_TO, MMP_COLNAME_MEANDIFF, MMP_COLNAME_DIFF, MMP_STRUCT_DIFF_FROM_NAME, MMP_STRUCT_DIFF_TO_NAME,
+  MMP_COLOR} from './mmp-constants';
+import {PaletteCodes} from './mmp-mol-rendering';
+
+function calculateActivityDiffs(molecules: DG.Column, activities: DG.ColumnList,
+  mmpRules: MmpRules, variates: number, allCasesNumber: number) :
+  { maxActs: number [],
```

Comment:
I would probably create an interface with all these fields, at least for the sake of documenting the fields :)